packages/core/test/scxml.test.ts

```diff
@@ -253,9 +253,9 @@ const testGroups = {
     'test387.txml',
     'test388.txml',
     'test396.txml',
-    // 'test399.txml', // inexact prefix event matching not implemented
-    // 'test401.txml', // inexact "error" event (should be "error.execution")
-    // 'test402.txml', // error.execution when evaluating assign + inexact prefix event matching not implemented
+    'test399.txml',
+    'test401.txml',
+    'test402.txml',
```

Comment:
I believe that this might be passing accidentally and this needs further investigation. It currently fails because context is undefined - I'm not 100% sure if that's correct. Maybe it should fail on assigning to non-existent context property (it tries to assign to `location="foo.bar.baz "`). Maybe failing  with:
```
Cannot assign to undefined `context`. Ensure that `context` is defined in the machine config.
```
is expected here but it's worth ensuring that - can be done later though.

If I unblock this by providing an empty context I hit another set of problems - this is why I believe it would be best to leave it as commented out for now.

1. Our script for [`exposing datamodel`](https://github.com/davidkpiano/xstate/blob/8d328e92f74d0a428aa66a2395f9aa85bbba3223/packages/core/src/scxml.ts#L107-L111) to expressions is naive - it results in things like: `const foo.bar.baz  = context['foo.bar.baz '];` which are syntax errors)
2. raised event is executed **before** executable content (`assign` in this test). This is not correct - `raise` should add to the internal queue from which the event should be taken off after executing whole executable content caused by a transition

```suggestion
    // 'test402.txml', // TODO: investigate more, it expects error.execution when evaluating assign, check if assigning to a deep location is even allowed, check if assigning to an initialized datamodel is allowed, improve how datamodel is exposed to constructed functions
```